index.html

```diff
@@ -862,35 +867,28 @@ <h3>Extensions to the <code>Performance</code> Interface</h3>
 buffer</a>.</li>
 <li>Increment <a>resource timing buffer current size</a> by 1.</li>
 <li>Remove <i>entry</i> from <a>secondary entry buffer</a>.</li>
+<li>Decrement <a>resource timing secondary buffer current size</a> by 1.</li>
 </ol>
-<li>Assert that <a>secondary entry buffer</a> is empty.</li>
 </ol>
 
 <p>To <dfn>fire a buffer full event</dfn>, run the following steps:</p>
 <ol>
-<li><a data-cite="DOM/#concept-event-fire">Fire an event</a>
-named <code>resourcetimingbufferfull</code> at the <a data-cite=
-"hr-time-2/#idl-def-performance">Performance</a> object,</li>
-<li>Run <a>copy secondary buffer</a>.</li>
-<li>While <a>resource timing buffer current size</a> is larger than 
-<a>resource timing buffer size limit</a>, run the following substeps:
+<li>While <a>resource timing buffer current size</a> equals
```

Comment:
This indicates that `resourcetimingbufferfull` will not be fired if the buffer is cleared between the time we queue the task and the time we run the task. This seems reasonable to me.

Reply:
Oh but actually there does not seem to be a guarantee that secondary buffer will be empty by the end of this algorithm? I think this task should be roughly as follows:
`While primary full OR secondary buffer not empty:`
`if primary full then fire resourcetimingbufferfull`
`if secondary nonempty then copy secondary. if nothing copied, empty secondary.`

Reply:
Good catch.

Reply:
Well, trying it out, the above algorithm ends up with an infinite loop, as the primary can be full but with no excess entries, in which case, no event should fire. Will fix.

Reply:
With the tests to keep me honest, I ended up with something like: "While secondary buffer is not empty: if primary full, then fire resourcetimingbufferfull; Copy secondary; If secondary is same size as before, empty secondary and abort"

Reply:
To clarify: The difference between "if nothing copied" and "same size as before" is for cases where resources where copied, but the same amount of resources was added inside the call. "if nothing copied" opens up a potential for infinite loops when the handler increases by one but also triggers a sync resource call at the same time.

Reply:
Ok looks much better! However, I still think "nothing copied" is slightly better here. If nothing copied, we can empty the secondary buffer and abort. Since the while now only depends on secondary buffer being nonempty, that will not cause infinite loops.

Reply:
It will if every event that fires adds an entry to the secondary buffer.  That way, the sync event loop will continue forever, because every iteration copies a single entry. Don't know how realistic that scenario is, but overall doesn't seem worth the risk.